### PR TITLE
Remove #if SHELL_ENABLE_METAL checks in iOS code

### PR DESCRIPTION
**Pull request by jmagman**
All physical iOS devices Flutter supports (iOS 12+) can run Metal.  The only time Flutter doesn't use Metal afaik is for iOS simulators running < iOS 13, which is covered by a few `if (@available(iOS METAL_IOS_VERSION_BASELINE, *))` checks.

https://github.com/flutter/engine/blob/8a51e124fbf168295a1b05f8d66459bfb29ae8a9/shell/platform/darwin/ios/rendering_api_selection.h#L37-L41

Remove hardware checks for physical devices.

Remove `shell_enable_metal` from the gn files and the `#if SHELL_ENABLE_METAL` checks in the iOS embedder.

I limited this PR to just iOS, but I imagine it's safe to remove `shell_enable_metal` everywhere? 
https://github.com/flutter/engine/blob/41da00ac46bc38a97a63ed0635450271d71afd9f/shell/platform/darwin/macos/BUILD.gn#L18
https://github.com/flutter/engine/blob/41da00ac46bc38a97a63ed0635450271d71afd9f/tools/gn#L673-L679

### CODE REVIEW

**Comment by cbracken** [comment]
> I imagine it's safe to remove shell_enable_metal everywhere?

I removed GL support from the macOS embedder and we don't have the software compositing fallback in that embedder.

Does the iOS embedder still support software rendering? I know we used use it for simulators, but IIRC those moved to Metal. Or was simulator Metal adoption limited to arm64 macs?

File: shell/platform/darwin/ios/ios_context.mm [hunk 1]
```diff
@@ -6,12 +6,9 @@
 #include "flutter/shell/platform/darwin/ios/rendering_api_selection.h"
 
 #include "flutter/fml/logging.h"
-#include "flutter/shell/platform/darwin/ios/ios_context_software.h"
-
-#if SHELL_ENABLE_METAL
 #include "flutter/shell/platform/darwin/ios/ios_context_metal_impeller.h"
 #include "flutter/shell/platform/darwin/ios/ios_context_metal_skia.h"
-#endif  // SHELL_ENABLE_METAL
+#include "flutter/shell/platform/darwin/ios/ios_context_software.h"
```
Comment (cbracken) [inline shell/platform/darwin/ios/ios_context.mm:11]: Curious what this is still used for.

**Comment by jmagman** [comment]
> I know we used to use it for simulators, but IIRC those moved to Metal. Or was simulator Metal adoption limited to arm64 macs?

The only time Flutter doesn't use Metal afaik is for iOS simulators running < iOS 13 #17881.  However, I think to run an iOS 12 simulator you need an older version of Xcode than the current one, which itself needs to be run on an older version of macOS.  It is still possible though with that setup.

https://github.com/flutter/engine/blob/8a51e124fbf168295a1b05f8d66459bfb29ae8a9/shell/platform/darwin/ios/rendering_api_selection.h#L37-L41

**Comment by jmagman** [comment]
> The only time Flutter doesn't use Metal afaik is for iOS simulators running < iOS 13 #17881.

Oh also when software rendering is "forced" for simulators, forgot about that.  Used for... scubas?

File: shell/platform/darwin/ios/ios_context.mm (hunk 1 above)
Comment (jmagman) [reply shell/platform/darwin/ios/ios_context.mm:11]: `IOSContextSoftware`
https://github.com/flutter/engine/blob/03cd54d4285a44e1e082ba0d1277fbee5b3f4f48/shell/platform/darwin/ios/ios_context.mm#L34C7-L34C53

**Comment by jmagman** [comment]
> I'd think we should be able to kill off the software rasterisation fallback

I think we may need to continue to support softrware rendering until the Skia fallback is removed in favor of Impeller. #44346
https://github.com/flutter/engine/blob/03cd54d4285a44e1e082ba0d1277fbee5b3f4f48/shell/platform/darwin/ios/ios_context.mm#L29-L33

@bdero is that a true statement?

**Comment by cbracken** [comment]
> I think to run an iOS 12 simulator you need an older version of Xcode than the current one, which itself needs to be run on an older version of macOS. It is still possible though with that setup.

Is that still a supported deployment target? If not, I'm all for scrub scrub scrub.

> Oh also when software rendering is "forced" for simulators, forgot about that. Used for... scubas?

It's possible/likely it's changed since 2016 when @goderbauer and I wrote the original scuba support but I think at that point we were using the exact same underlying mechanism for screenshotting the tool uses -- i.e. wiring through the call to `[NSView drawLayer:inContext:]`. 

But now that you mention it, maybe we did force software rasterisation? Something about reproducible results? i.e. the same reason we run our GPU screenshot tests through SwiftShader i engine tests? The sad part is I think we'd need to work that out from the code, cause 2016 is definitely further back than I clearly remember, and either way, so much has changed that even if I could, the chance it still applies is near 0%.

**Comment by jmagman** [comment]
> Is that still a supported deployment target? If not, I'm all for scrub scrub scrub.

Technically yes, iOS 12 is the minimum supported.  Though it is really tricky getting that simulator installed. We can rip out support next year when Apple (likely) drops support for it.

File: shell/platform/darwin/ios/rendering_api_selection.mm [hunk 2]
```diff
@@ -18,18 +16,6 @@
 
 namespace flutter {
 
-#if SHELL_ENABLE_METAL
-bool ShouldUseMetalRenderer() {
```
Comment (jonahwilliams) [inline shell/platform/darwin/ios/rendering_api_selection.mm:22]: Removing this is likely what broke the google3 test harness. Despite running on a version of iOS sim that is *required* to support metal - due to the lack of supported GPU passthrough (and in defiance of all that is holu), creation of the MTLDevice will fail.

Comment (jmagman) [reply shell/platform/darwin/ios/rendering_api_selection.mm:22]: Makes sense, I'll fix it up. Thank you for looking at it!

**Pull request merged**